pkg/config/storage.go

```diff
@@ -0,0 +1,102 @@
+package config
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// DriverOpt is an enum value that determines whether MicroShift deploys LVMS.
+type DriverOpt string
+
+const (
+	// DriverUnset exists to support backwards compatibility with existing MicroShift clusters. When .storage.driver is
+	// "", MicroShift will default to deploying LVMS. This preserves the current deployment behavior of existing
+	// clusters.
+	DriverUnset DriverOpt = ""
+	//  DriverNone signals MicroShift to not deploy the LVMS components. Setting the value for a cluster that has already
+	//  deployed LVMS will not cause LVMS to be deleted. Otherwise, volumes already deployed on the cluster would be
+	//  orphaned once their workloads stop or restart.
+	DriverNone DriverOpt = "none"
+	// DriverLVMS is equivalent to DriverUnset, and explicitly tells MicroShift to deploy LVMS. This option exists to
+	// provide a differentiation between LVMS and potential future driver options.
+	DriverLVMS DriverOpt = "lvms"
+)
+
+// ComponentOpt values determine which CSI components MicroShift should deploy. Currently only csi snapshot components
+// are supported.
+type ComponentOpt string
+
+const (
+	// CsiComponentNone exists to support backwards compatibility with existing MicroShift clusters. By default,
+	// MicroShift will deploy snapshot controller and webhook when no components are specified. This preserves the
+	// current deployment behavior of existing clusters. Users must set .storage.with-csi-components: [ "none" ] to
+	// explicitly tell MicroShift not to deploy any CSI components. The CSI Driver is excluded as it is typically
+	// deployed via the same manifest as the accompanying storage driver. Like DriverOpt, uninstallation is not
+	// supported as this can lead to orphaned storage volumes. Mutually exclusive with all other ComponentOpt values.
+	CsiComponentNone ComponentOpt = "none"
+	// CsiComponentSnapshot causes MicroShift to deploy the CSI Snapshot controller.
+	CsiComponentSnapshot ComponentOpt = "snapshot-controller"
+	// CsiComponentSnapshotWebhook causes MicroShift to deploy the CSI Snapshot Validation Webhook.
+	CsiComponentSnapshotWebhook ComponentOpt = "snapshot-webhook"
+)
+
+// Storage represents a subfield of the microshift config data structure. Its purpose to provide a user
+// facing interface to control whether MicroShift should deploy LVMS on startup.
+type Storage struct {
+	Driver        DriverOpt      `json:"driver,omitempty"`
+	CSIComponents []ComponentOpt `json:"with-csi-components,omitempty"`
```

Comment:
We have camel case in every other parameter in the config, should we follow it here too?